src/components/Premium/Premium.tsx

```diff
@@ -4,44 +4,92 @@ import Payment from "../Payment/Payment";
 import './Premium.css';
 import { wasAlreadyLoggedIn } from '../../utils/GoogleUtils';
 import { getLoadingElement } from "../../utils/LoadingUtils";
-import { Card } from "react-bootstrap";
+import { Card, Modal, OverlayTrigger, Tooltip } from "react-bootstrap";
 import NavBar from "../NavBar/NavBar";
+import PremiumFeatures from "./PremiumFeatures/PremiumFeatures";
+import { Link as LinkIcon } from '@material-ui/icons';
+import api from "../../api/ApiHelper";
+import moment from 'moment';
+import { v4 as generateUUID } from 'uuid';
 
 let wasAlreadyLoggedInGoogle = wasAlreadyLoggedIn();
 
 function Premium() {
 
     let [isLoggedIn, setIsLoggedIn] = useState(false);
+    let [showFeatureDialog, setShowFeatureDialog] = useState(false);
+    let [hasPremium, setHasPremium] = useState(false);
+    let [hasPremiumUntil, setHasPremiumUntil] = useState<Date | undefined>();
+
+    function loadHasPremiumUntil(): Promise<void> {
+        let googleId = localStorage.getItem('googleId');
+        return api.hasPremium(googleId!).then((hasPremiumUntil) => {
+            if (hasPremiumUntil) {
+                setHasPremium(true);
+            }
+            setHasPremiumUntil(hasPremiumUntil)
+        });
+    }
+
 
     function onLogin() {
         let googleId = localStorage.getItem('googleId');
         if (googleId) {
             setIsLoggedIn(true);
+            loadHasPremiumUntil();
         }
     }
 
+    function closeFeatureDialog() {
+        setShowFeatureDialog(false);
+    }
+
+    let featureDialog = (
+        <Modal show={showFeatureDialog} onHide={closeFeatureDialog}>
+            <Modal.Header closeButton>
+                <Modal.Title>Features</Modal.Title>
+            </Modal.Header>
+            <Modal.Body>
+                <PremiumFeatures />
+            </Modal.Body>
+        </Modal>
+    );
+
     return (
         <div className="premium">
-            <h4>
+            <h2>
                 <NavBar />
                 Premium
-            </h4>
+            </h2>
+            <hr />
+            {hasPremium ? <p style={{ color: "#00bc8c" }}>You have a premium account. Thank you for your support.</p> : <p style={{ color: "red" }}>You do no have a premium account</p>}
```

Comment:
Please also display the Google user name and email.
You get those from the middle part of the google token. (its separated by dots and bas64 encoded)